compiler/src/main/kotlin/ast/symbol.kt

```diff
@@ -0,0 +1,82 @@
+package dk.aau.cs.d409f19.cellumata.ast
+
+import java.util.*
+
+data class Table(
+    val symbols: MutableMap<String, AST> = mutableMapOf(),
+    val tables: MutableList<Table> = mutableListOf()
+)
+
+class SymbolRedefinitionException(val ident: String) : Exception("\"$ident\" is already defined")
+
+private val RESERVED_SYMBOLS: List<String> = listOf(
+    "world",
+    "neighbourhood",
+    "state",
+    "function",
+    "int",
+    "float",
+    "bool",
+    "become",
+    "return",
+    "if",
+    "elif",
+    "else",
+    "let",
+    "for",
+    "continue",
+    "break",
+    "rand",
+    "abs",
+    "floor",
+    "ceil",
+    "sqrt",
+    "pow"
```

Comment:
I think we should take care of this in a future pull request since resolving it is a rather complex matter involving how we should handle function overloading.

Reply:
In that case, unless @NicEastvillage disagrees, we should make an issue to track the work.